### ISSUE TITLE
Minimum training step and loss calculation frequency not documented

### ISSUE THREAD

**Issue by NickBear-star** [tensorflow/models#10578]
# Prerequisites

Please answer the following question for yourself before submitting an issue.

- [x] I checked to make sure that this issue has not been filed already.

## 1. The entire URL of the documentation with the issue

- https://github.com/tensorflow/models/tree/master/research/object_detection/protos
- https://github.com/tensorflow/models/blob/master/research/object_detection/model_main_tf2.py

## 2. Describe the issue

I tried to run 20 training steps, with a loss calculation every 2 steps with the OD API, using the following command line:

```Bash
model_main_tf2.py
  --model_dir="/this/directory/"
  --pipeline_config_path="/this/pipeline.config"
  --log_dir="/this/log/directory"
  --num_train_steps=20
  --num_eval_steps=2
  --checkpoint_every_n=2
  --eval_timeout=30
  --verbose=0 
```

However, the training event file only contains the loss calculated at step 100 (there is nothing in the log directory). I modified the pipeline.config file as following:

```
[...]
train_config {
  [...]
  sync_replicas: false
  num_steps: 20
  [...]
}
[...]
```

The result was the same however: 100 training steps with a single loss calculation at step 100 on the training set. This let me suppose that there is a minimum to the number of training step (100 steps?) and that the loss calculation frequency (100 steps?) cannot be tuned. This should to be clearly stated in the documentation of proto files and `model_main_tf2.py`.
